dm/worker/server.go

```diff
@@ -142,16 +146,21 @@ func (s *Server) Start() error {
 		defer s.wg.Done()
 		// TODO: handle fatal error from observeSourceBound
 		//nolint:errcheck
-		s.observeSourceBound(s.ctx, revBound)
+		for {
+			s.observeSourceBound(s.ctx, revBound)
```

Comment:
we might make use of the returned error (for example, if `err == nil` means parent context is done)

Reply:
please address above comment